dali/test/python/reader/test_fits.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2020-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali import Pipeline
+import nvidia.dali.fn as fn
+import os
+from astropy.io import fits
+import numpy as np
+import tempfile
+import random
+from test_utils import to_array
+from numpy.testing import assert_array_equal
+
+rng = np.random.RandomState(12345)
+
+
+def create_fits_file(filename, shape, type=np.int32, compressed=False):
+    imageData = rng.randint(100, size=shape).astype(type)
+    imageHeader = fits.Header()
+    hdu = fits.ImageHDU(imageData, imageHeader)
+    if compressed:
+        hdu = fits.CompImageHDU(imageData, imageHeader)
+
+    hdu.writeto(filename)
+
+
+def delete_fits_file(filename):
+    if os.path.isfile(filename):
+        os.remove(filename)
```

Comment:
Unused, you could probably take a look into setting up a Unittest setUp and tearDown functions for this purpose.